src/main/java/org/broadinstitute/hellbender/tools/dataflow/transforms/bqsr/BaseRecalibratorFn.java

```diff
@@ -333,11 +259,22 @@ protected boolean badSolidOffset(final GATKRead read, final int offset) {
         return knownSites;
     }
 
+
+    private boolean isLowQualityBase(final GATKRead read, final int offset) {
+        return read.getBaseQualities()[offset] < minimumQToUse;
+    }
+
+    private boolean badSolidOffset(final GATKRead read, final int offset) {
+        return ReadUtils.isSOLiDRead(read, header) && BRAC.RAC.SOLID_RECAL_MODE != RecalUtils.SOLID_RECAL_MODE.DO_NOTHING && !RecalUtils.isColorSpaceConsistent(read, offset);
+    }
+
     // TODO: can be merged with calculateIsIndel
-    protected static int[] calculateIsSNP(final GATKRead read, /*final BasesCache ref*/ final ReferenceContext ref, final GATKRead originalRead) {
+    private static int[] calculateIsSNP(final GATKRead read, final ReferenceContext ref, final GATKRead originalRead) {
         final byte[] readBases = read.getBases();
         final byte[] refBases = Arrays.copyOfRange(ref.getBases(), read.getStart() - originalRead.getStart(), ref.getBases().length + read.getEnd() - originalRead.getEnd());
-        int refPos = 0;
+        // the referenceContext may start a little before the read; correct for that.
```

Comment:
Rephrase this: It **will** always start before the start of the read (because of the way the BQSR_WINDOW_FUNCTION works by fetching extra reference context for BAQ)